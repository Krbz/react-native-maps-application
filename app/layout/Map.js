import React, { Component } from "react";
import { connect } from "react-redux";
import { styles } from "../styles.android";

import MapView from "react-native-maps";
import { View } from "react-native";
import { bindActionCreators } from "redux";
import * as drawerActionsCreators from "../actions/drawerActions";
import BurgerMenuButton from "../components/BurgerMenu";

export class MapLayout extends Component {
  constructor(props) {
    super(props);

    this.state = {
      region: {
        latitude: 52.2300,
        longitude: 21.0000,
        latitudeDelta: 0.0922,
        longitudeDelta: 0.0421,
      },
      drawer: {
        visible: false,
      },
    };

    this.showDrawer = this.showDrawer.bind(this);
  }

  showDrawer() {
    const prepare = Object.assign({}, this.state.drawer, { visible: true });

    this.props.drawerActions.showDrawer(prepare);
  }

  render() {
    return (
      <View style={styles.mapLayout}>
        <MapView
          style={styles.map}
          initialRegion={this.state.region}/>

        <View style={styles.mapOverLayer}>
          <BurgerMenuButton onPress={this.showDrawer} />
        </View>
      </View>
    );
  }
}

/* istanbul ignore next */
function mapStateToProps(state) {
  return {
    drawerReducer: state.drawerReducer,
  }
}

/* istanbul ignore next */
function mapDispatchToProps(dispatch) {
  return {
    drawerActions: bindActionCreators(drawerActionsCreators, dispatch),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MapLayout)
