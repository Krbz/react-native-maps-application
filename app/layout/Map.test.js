import "react-native";
import React from "react";

import renderer from "react-test-renderer";
import { MapLayout } from "./Map";

test('renders correctly', () => {
  const tree = renderer.create(
    <MapLayout />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});