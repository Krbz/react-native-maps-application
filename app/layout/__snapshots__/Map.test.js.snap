// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#eeeeee",
    }
  }
>
  <View />
</RCTScrollView>
`;
